tests/test_decorator.py

```diff
@@ -160,5 +160,40 @@ def simple_func(session: CoveSession) -> Optional[str]:
         return session.session_information.RoleSessionName
 
     cove_output = simple_func()
-    print(cove_output)
+
+    assert cove_output["Exceptions"] == []
```

Comment:
The extra assertion is so that the test doesn't pass when an exception was actually raised and the length of Results is 0.

My tests were reporting success even though they were really failing. The hidden error was this:

> 'CoveSessionInformation' object is not subscriptable

Because I hadn't rewritten the test code correctly.

It was actually the flakehell linter that caught this indirectly by warning my about a variable name that I had forgotten to update for the new test. I realised that if the linter could detect it, then the test should have failed, but it was gladly passing.

Reply:
Good catch, probably a dodgy use of all() on my part here